pkg/icingadb/sync.go

```diff
@@ -160,9 +182,52 @@ func (s Sync) ApplyDelta(ctx context.Context, delta *Delta) error {
 	// Delete
 	if len(delta.Delete) > 0 {
 		s.logger.Infof("Deleting %d items of type %s", len(delta.Delete), utils.Key(utils.Name(delta.Subject.Entity()), ' '))
+		entity := delta.Subject.Entity()
+		if _, ok := entity.(*v1.Service); ok {
+			s.logger.Infof("Inserting %d items of type service sla history trails of type delete", len(delta.Delete))
+			g.Go(func() error {
+				columns := &SlaServiceHistoryTrailColumns{}
+				query := s.db.BuildSelectStmt(entity, columns)
+				if len(delta.Delete) == 1 {
+					query += ` WHERE id = ?`
+				} else {
+					var placeholders []string
+					for i := 0; i < len(delta.Delete); i++ {
+						placeholders = append(placeholders, "?")
+					}
+
+					query += fmt.Sprintf(` WHERE id IN (%s)`, strings.Join(placeholders, `, `))
+				}
+				entities, err := s.db.YieldAll(ctx, delta.Subject.Factory(), query, false, delta.Delete.IDs()...)
+				com.ErrgroupReceive(g, err)
+
+				return s.db.CreateStreamed(ctx, CheckableToSlaTrailEntities(ctx, g, entities, "delete"))
+			})
+		}
+
+		var hostIds chan any
+		onSuccessHandlers := []OnSuccess[any]{OnSuccessIncrement[any](stat)}
+
+		_, isHost := entity.(*v1.Host)
```

Comment:
... a switch/case IMAO.

Reply:
It won't make any difference as the host IDs goroutine is dependent on the `onSuccess` handler of the Delete goroutine.

Reply:
NM!